src/mpid/ch4/netmod/ofi/ofi_types.h

```diff
@@ -397,6 +350,23 @@ typedef struct {
     char kvsname[MPIDI_KVSAPPSTRLEN];
     char pname[MPI_MAX_PROCESSOR_NAME];
     int port_name_tag_mask[MPIR_MAX_CONTEXT_MASK];
+
+    /* Capability settings */
+#ifdef MPIDI_OFI_ENABLE_RUNTIME_CHECKS
```

Comment:
That's true, but what you get back may not be optimal. A single provider can support regular endpoints and scalable endpoints. `AV_MAP` and `AV_TABLE`. The point is that if you don't specifically pick all of them, you don't know what you'll get. It may or may not match up with what we ask for in the pre-filled capability set.

Here's a concrete example. Let's say that you really want regular endpoints instead of scalable endpoints for some reason, but you don't care about the rest of the capabilities. So your capability set would look like this:

```
/* We care about this one so set it to 0. */
enable_scalable_endpoints = 0;

/* We don't care about these so set them to the default (1).
enable_data = 1;
enable_av_table = 1;
enable_stx_rma = 1;
enable_mr_scalable = 1;
enable_tagged = 1;
enable_am = 1;
enable_rma = 1; 
```

If the provider you're looking for doesn't meet these requirements, there's no way to say, give me regular endpoints and I don't care about the rest. You might get back no providers from `fi_info`.

The problem isn't when you select a provider. In that case, there's no runtime checks (that's something we could add, the ability to both select a provider and have runtime configuration). The problem is when you select runtime (or don't pick anything), but still want to tweak a few things, we don't know good defaults for the rest.

Does that make sense? Probably the best thing would be to add the feature that I mentioned in the middle. Let you both select a provider and pick the runtime capabilities while pre-filling the rest.